### PR TITLE
Use ActiveRecord validations for user creation

### PR DESCRIPTION
**Pull request by duckth**
This PR changes the register endpoints to use the native ActiveRecord validation methods instead of the "validate_create_user" endpoint. The messages have been changed in the tests to reflect the messages ActiveRecord generates. Shouldn't affect the simulator as it doesn't actually care about the message.

### CODE REVIEW

**Review by silkeholmebonnen — DISMISSED** [review]
LGTM

File: myapp.rb [hunk 1]
```diff
@@ -98,28 +94,19 @@ def current_user
...
-  elsif nil_or_empty?(params[:email]) || params[:email] !~ /@/
-    error = 'You have to enter a valid email address'
-  elsif nil_or_empty?(params[:password])
-    error = 'You have to enter a password'
-  elsif params[:password] != params[:password2]
-    error = 'The two passwords do not match'
-  elsif User.find_by_username(params[:username])
-    error = 'The username is already taken'
-  elsif User.create(
+  u = User.new(
```
Comment (fredpetersen) [inline myapp.rb:97]: I'd prefer for this to be completely explicit and be called user instead

**Review by fredpetersen — COMMENTED** [review]
As long as the simulator doesn't expect specific error messages this looks fine to be

**Review by fredpetersen — APPROVED** [review]
All good

**Review by MadsRoager — APPROVED** [review]
Nice

**Review by A-Guldborg — APPROVED** [review]

**Pull request merged**